### PR TITLE
Peter/key get

### PR DESCRIPTION
**Pull request by kingofirony**
- Gets keys from server

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,6 +1,8 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
-    package="com.mapzen.erasermap">
+    package="com.mapzen.erasermap"
+    android:versionCode="2"
+    android:versionName="0.1">
```
Comment (ecgreb) [inline app/src/main/AndroidManifest.xml]: No need to declare version code and version name here. It gets injected by Gradle at build time.

File: app/src/main/java/com/mapzen/erasermap/CommonModule.java [hunk 2]
```diff
@@ -26,7 +27,9 @@
         return new Bus();
     }
 
-    @Provides RouteEngine provideRouteEngine() {
+    @Provides @Singleton ManifestModel provideApiKeys() { return new ManifestModel(); }
```
Comment (ecgreb) [inline app/src/main/java/com/mapzen/erasermap/CommonModule.java]: Typically Dagger provider methods will mirror name of return type. For example `provideManifestModel()`.

File: app/src/main/java/com/mapzen/erasermap/CommonModule.java [hunk 3]
```diff
@@ -26,7 +27,9 @@
...
+
+    @Provides RouteEngine provideRouteEnCgine() {
```
Comment (ecgreb) [inline app/src/main/java/com/mapzen/erasermap/CommonModule.java]: Typo

File: app/src/main/kotlin/com/mapzen/erasermap/view/MainActivity.kt [hunk 4]
```diff
@@ -90,6 +93,7 @@ public class MainActivity : AppCompatActivity(), MainViewController, Router.Call
         initReverseButton()
         centerMapOnCurrentLocation()
         presenter?.onRestoreViewState()
+        getApiKeys({initCrashReportService()})
```
Comment (ecgreb) [inline app/src/main/kotlin/com/mapzen/erasermap/view/MainActivity.kt]: Does this mean initializing crash reports doesn't happen until the API manifest is downloaded? What if the app crashes before the download completes? Or the download itself crashes the app? I think we would still want to be able to catch those errors.

Perhaps we want to continue to inject the MINT_API_KEY via BuildConfig and reserve the API manifest for our services (Vector Tiles, Pelias, Valhalla).

File: app/src/main/kotlin/com/mapzen/erasermap/view/MainActivity.kt [hunk 5]
```diff
@@ -65,6 +67,8 @@ public class MainActivity : AppCompatActivity(), MainViewController, Router.Call
       @Inject set
     var crashReportService: CrashReportService? = null
       @Inject set
+    var apiKeys: ManifestModel? = null
+      @Inject set
```
Comment (ecgreb) [inline app/src/main/kotlin/com/mapzen/erasermap/view/MainActivity.kt]: Do we need to inject an empty `ManifestModel` here when a new instance is created by `ManifestDownloader`?

File: app/src/test/java/com/mapzen/erasermap/model/ManifestDownLoaderTest.kt [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+    }
+
+    After
+    throws(Exception::class)
+    public fun teardown() {
+        server?.shutdown()
+    }
+
+    Test
```
Comment (ecgreb) [inline app/src/test/java/com/mapzen/erasermap/model/ManifestDownLoaderTest.kt]: Wow the `@` symbol is optional for JUnit annotations... weird. Nice job getting tests to run in Kotlin.

Comment (kingofirony) [inline app/src/test/java/com/mapzen/erasermap/model/ManifestDownLoaderTest.kt]: Done

File: app/src/main/java/com/mapzen/erasermap/CommonModule.java (hunk 2 above)
Comment (kingofirony) [inline app/src/main/java/com/mapzen/erasermap/CommonModule.java]: Done

File: app/src/main/java/com/mapzen/erasermap/CommonModule.java (hunk 3 above)
Comment (kingofirony) [inline app/src/main/java/com/mapzen/erasermap/CommonModule.java]: Done

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (kingofirony) [inline app/src/main/AndroidManifest.xml]: Done

File: app/src/main/kotlin/com/mapzen/erasermap/view/MainActivity.kt (hunk 4 above)
Comment (kingofirony) [inline app/src/main/kotlin/com/mapzen/erasermap/view/MainActivity.kt]: Done

File: app/src/main/kotlin/com/mapzen/erasermap/view/MainActivity.kt (hunk 5 above)
Comment (kingofirony) [inline app/src/main/kotlin/com/mapzen/erasermap/view/MainActivity.kt]: Done

**Pull request merged**